DXRVanilla/DeusEx/Classes/PersonaScreenInventory.uc

```diff
@@ -129,3 +134,146 @@ event DescendantRemoved(Window descendant)
+	btnDrop.SetButtonText(DropButtonLabel);
+
+	btnUse = PersonaActionButtonWindow(winActionButtons.NewChild(Class'PersonaActionButtonWindow'));
+	btnUse.SetButtonText(UseButtonLabel);
+
+	btnEquip = PersonaActionButtonWindow(winActionButtons.NewChild(Class'PersonaActionButtonWindow'));
+	btnEquip.SetButtonText(EquipButtonLabel);
+}
+
+function EnableButtons()
+{
+    local DataStorage datastorage;
+    local Inventory inv;
+
+	if ((btnRefusal == None) || (btnDrop == None) || (btnEquip == None) || (btnUse == None))
+		return;
+
+    btnRefusal.SetButtonText(RefuseLabel);
+
+	if (selectedItem == None) {
+		btnDrop.DisableWindow();
+		btnRefusal.DisableWindow();
+		btnEquip.DisableWindow();
+		btnUse.DisableWindow();
+	} else {
+		btnEquip.EnableWindow();
+		btnUse.EnableWindow();
+		btnDrop.EnableWindow();
+		btnRefusal.EnableWindow();
+
+		inv = Inventory(selectedItem.GetClientObject());
+
+		if (inv != None) {
+            datastorage = class'DataStorage'.static.GetObj(class'DXRando'.default.dxr);
+            if (InStr(datastorage.GetConfigKey("item_refusals"), "," $ inv.class.name $ ",") == -1) {
+                btnRefusal.SetButtonText(RefuseLabel);
+            } else {
+                btnRefusal.SetButtonText(AcceptLabel);
+            }
+
+            if (NanoKeyRing(inv) != None) {
+                btnDrop.DisableWindow();
+                btnRefusal.DisableWindow();
+				btnEquip.DisableWindow();
+				btnUse.DisableWindow();
+			} else {
+                if (WeaponMod(inv) != None || AugmentationUpgradeCannister(inv) != None) {
+                    btnUse.DisableWindow();
+                } else {
+                    if ((inv == player.inHand ) || (inv == player.inHandPending))
+                        btnEquip.SetButtonText(UnequipButtonLabel);
+                    else
+                        btnEquip.SetButtonText(EquipButtonLabel);
+                }
+            }
+            log("debug inv: " $ inv);
+		} else {
+			btnDrop.DisableWindow();
+		    btnRefusal.DisableWindow();
+			btnEquip.DisableWindow();
+			btnUse.DisableWindow();
+		}
+	}
+}
+
+function bool ButtonActivated(Window buttonPressed)
+{
+    local bool ret;
+
+    if (buttonPressed == btnRefusal) {
+        Refuse();
+        return true;
+    }
+
+    ret = Super.ButtonActivated(buttonPressed);
+    if (PersonaAmmoDetailButton(buttonPressed) != None) {
+        btnDrop.DisableWindow();
+        btnRefusal.DisableWindow();
+        btnEquip.DisableWindow();
+        btnUse.DisableWindow();
+    }
+    return ret;
+}
+
+function Refuse()
+{
+    local DataStorage datastorage;
+    local Inventory item;
+    local Pickup droppedItem;
+    local string item_refusals, leftPart, rightPart;
+    local int strIdx;
+    local Vector x, y, z;
+    local Vector dropVect;
+
+    datastorage = class'DataStorage'.static.GetObj(class'DXRando'.default.dxr);
+    item = Inventory(selectedItem.GetClientObject());
+
+    item_refusals = datastorage.GetConfigKey("item_refusals");
```

Comment:
actually maybe all this logic belongs in DXRLoadouts, along with the IsRefused function